libbeat/cmd/instance/beat.go

```diff
@@ -864,6 +865,19 @@ func (b *Beat) configure(settings Settings) error {
 		version.SetPackageVersion(b.Info.Version)
 	}
 
+	// if we're in fleet mode, construct a custom user-agent
+	if fleetmode.Enabled() {
+		userAgent := ""
+		info := b.Manager.AgentInfo()
+		if info.ManagedMode == proto.AgentManagedMode_MANAGED {
+			userAgent = "Agent-Managed"
+		} else if info.ManagedMode == proto.AgentManagedMode_STANDALONE {
+			userAgent = "Agent-Standalone"
+		}
+		b.Info.UserAgentPostfix = userAgent
+	}
+	logp.L().Infof("got user agent in beat: %s", b.Info.UserAgentPostfix)
```

Comment:
This should probably be a debug log. We are trying to minimize the amount the Beats log at startup to reduce the log volume in the agent monitoring indices.